src/types/multiroot.rs

```diff
@@ -0,0 +1,286 @@
+    pub fn set<F: Fn(& ::VectorF64, &mut ::VectorF64)-> ::Value>(
+        &mut self,
+        f: F,
+        n: usize,
+        x: &::VectorF64,
+    ) -> ::Value {
+
+        unsafe extern "C" fn inner_f<A: Fn(& ::VectorF64, &mut ::VectorF64)->::Value>(
+            x: *const sys::gsl_vector,
+            params: *mut ::std::os::raw::c_void,
+            f: *mut sys::gsl_vector,
+        ) -> std::os::raw::c_int {
+            let g: &A = &*(params as *const A);
+            let x_new = ::VectorF64::soft_wrap(x as *const _ as * mut _);
+            ::Value::into(g(&x_new, &mut ::VectorF64::soft_wrap(f)))
+        }
+
+        self.inner_call = sys::gsl_multiroot_function_struct {
+            f: Some(inner_f::<F>),
+            n,
+            params: &f as *const _ as *mut _,
+        };
+
+        // self.inner_call = wrap_callback_multi!(f, F, n);
+
+        ::Value::from(unsafe {
+            sys::gsl_multiroot_fsolver_set(
+                self.unwrap_unique(),
+                &mut self.inner_call,
+                x.unwrap_shared(),
+            )
+        })
+    }
+
+    /// This function performs a single iteration of the minimizer s. If the iteration encounters an
+    /// unexpected problem then an error code will be returned,
+    ///
+    /// ::Value::BadFunc
+    /// the iteration encountered a singular point where the function evaluated to Inf or NaN.
+    ///
+    /// ::Value::Failure
+    /// the algorithm could not improve the current best approximation or bounding interval.
+    ///
+    /// The minimizer maintains a current best estimate of the position of the minimum at all times,
+    /// and the current interval bounding the minimum. This information can be accessed with the
+    /// following auxiliary functions,
+    #[doc(alias = "gsl_min_fminimizer_iterate")]
+    pub fn iterate(&mut self) -> ::Value {
+        ::Value::from(unsafe { sys::gsl_multiroot_fsolver_iterate(self.unwrap_unique()) })
+    }
+
+    /// This function returns the current estimate of the root for the solver `s`, given by `s->x`.
+    pub fn root(& self) -> ::VectorF64 {
+        ::VectorF64::soft_wrap(unsafe {sys::gsl_multiroot_fsolver_root(self.unwrap_shared())})
+    }
+
+    /// This function returns the last step `dx` taken by the solver `s`, given by `s->dx`.
+    pub fn dx(& self) -> ::VectorF64 {
+        ::VectorF64::soft_wrap(unsafe {sys::gsl_multiroot_fsolver_dx(self.unwrap_shared())})
+    }
+
+    /// This function returns the function value `f(x)` at the current estimate of the root for
+    /// the solver `s`, given by `s->f`.
+    pub fn f(& self) -> ::VectorF64 {
+        ::VectorF64::soft_wrap(unsafe {sys::gsl_multiroot_fsolver_f(self.unwrap_shared())})
+    }
+
+}
+
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use VectorF64;
+    use multiroot::test_residual;
+
+    /// checking a test function
+    /// must return a success criteria (or failure)
+    fn rosenbrock_f(x: & ::VectorF64, f: &mut ::VectorF64) -> ::Value{
```

Comment:
I'm surprised `cargo fmt` didn't catch this one...

```suggestion
    fn rosenbrock_f(x: &::VectorF64, f: &mut ::VectorF64) -> ::Value{
```